ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetadataReceivedListenerImpl.java

```diff
@@ -0,0 +1,254 @@
+     */
+    private void restoreMetaData(final String objectName, final Metadata metadata) {
+        final StringBuilder builder = new StringBuilder();
+        final Joiner joiner = Joiner.on(", ");
+
+        builder.append("Metadata for object ").append(objectName).append(": ");
+        final String os = System.getProperty("os.name");
+
+        String creationTime = null;
+        String accessTime = null;
+        String modifiedTime = null;
+        String owner = null;
+        String groupName = null;
+        String storedOS = null;
+
+        if (metadata.get("ds3-creationTime").size() > 0)
+            creationTime = metadata.get("ds3-creationTime").get(0);
+
+        if (metadata.get("ds3-accessTime").size() > 0)
+            accessTime = metadata.get("ds3-accessTime").get(0);
+
+        if (metadata.get("ds3-lastModifiedTime").size() > 0)
+            modifiedTime = metadata.get("ds3-lastModifiedTime").get(0);
+
+        if (modifiedTime != null && creationTime != null && accessTime != null) {
+            setFileTimes(localFilePath + "/" + objectName, creationTime, modifiedTime, accessTime);
+        }
+
+        if (metadata.get("ds3-owner").size() > 0)
+            owner = metadata.get("ds3-owner").get(0);
+
+        if (metadata.get("ds3-groupName").size() > 0)
+            groupName = metadata.get("ds3-groupName").get(0);
+
+        if (owner != null && groupName != null && !owner.equals("") && !groupName.equals(""))
+            setOwnerNGroup(localFilePath + "/" + objectName, owner, groupName);
+
+        if (metadata.get("ds3-os").size() > 0)
+            storedOS = metadata.get("ds3-os").get(0);
+
+        if (storedOS != null && storedOS.equals(os) && os.contains("Windows")) {
+            setPermissionsForWindows(metadata, objectName);
+        } else {
+            if (metadata.get("ds3-permisions").size() > 0) {
+                String permissions = metadata.get("ds3-permisions").get(0);
+                if (permissions != null && !permissions.equals("")) {
+                    final Matcher m = Pattern.compile("\\(([^)]+)\\)").matcher(permissions);
+                    while (m.find()) {
+                        permissions = (m.group(1));
+                    }
+                    setPermissions(localFilePath + "/" + objectName, permissions);
+                }
+            }
+        }
+
+
+        for (final String metadataKey : metadata.keys()) {
+            final List<String> values = metadata.get(metadataKey);
+            builder.append("<Key: ")
+                    .append(metadataKey)
+                    .append(" Values: ")
+                    .append(joiner.join(values))
+                    .append("> ");
+        }
+    }
+
+    private void setPermissionsForWindows(final Metadata metadata, final String objectName) {
+        final Path path = Paths.get(localFilePath + "/" + objectName);
+        final AclFileAttributeView view1 = Files.getFileAttributeView(path, AclFileAttributeView.class);
```

Comment:
Why is this named view1?  I don't see where there is a view2.  Also, rather than just naming this view, I would update it's name to `attributeView`.